neqo-crypto/src/aead.rs

```diff
@@ -123,14 +123,35 @@ impl RealAead {
                 c_uint::try_from(output.len())?,
             )
         }?;
-        Ok(&output[0..(l.try_into()?)])
+        Ok(&output[..l.try_into()?])
     }
 
-    /// Decrypt a ciphertext.
+    /// Encrypt `data` consisting of `aad` and plaintext `data` in place.
     ///
-    /// Note that NSS insists upon having extra space available for decryption, so
-    /// the buffer for `output` should be the same length as `input`, even though
-    /// the final result will be shorter.
+    /// The space provided in `data` needs to allow `Aead::expansion` more bytes to be appended.
+    ///
+    /// # Errors
+    ///
+    /// If the input can't be protected or any input is too large for NSS.
+    pub fn encrypt_in_place(&self, count: u64, aad: &[u8], data: &[u8]) -> Res<usize> {
```

Comment:
Given that `decrypt_in_place` returns `&'a mut [u8]`, should this do the same?

IMPORTANT: You need to do this, at a minimum.

```suggestion
    pub fn encrypt_in_place(&self, count: u64, aad: &[u8], data: &mut [u8]) -> Res<usize> {
```

Though with the above...


```suggestion
    pub fn encrypt_in_place<'a>(&self, count: u64, aad: &[u8], data: &'a mut [u8]) -> Res<&'a mut [u8]> {
```